onnxruntime/core/optimizer/layer_norm_fusion.cc

```diff
@@ -345,6 +345,18 @@ X --> Pow --> ReduceMean --> Add --> Sqrt --> Div --> Mul
 |                                              ^
 |                                              |
 +----------------------------------------------+
+Additional FP16 patterns supported if allow_precision_change_ is true:
+X --> Cast1 --> Pow --> ReduceMean --> Add --> Sqrt --> Div --> Cast2 --> Mul
```

Comment:
It is not clear (due to new and existing code) if the code is checkking-out whether any of the intermediate nodes have other outputs.

Reply:
The calls to optimizer_utils::CheckOutputEdges() do check that the nodes have either graph outputs or other node outputs. This check was missing at one place in the new code, which I have added now